nixos/modules/services/monitoring/prometheus/exporters/unbound.nix

```diff
@@ -0,0 +1,72 @@
+{ config, lib, pkgs, options }:
+
+with lib;
+
+let
+  cfg = config.services.prometheus.exporters.unbound;
+in
+{
+  port = 9167;
+  extraOpts = {
+    fetchType = mkOption {
+      type = types.enum [ "shm" "tcp" "uds" ];
+      default = "shm";
+      description = ''
+        Which methods the exporter uses to get the information from unbound.
+      '';
+    };
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = ''
+        Address under which the exporter exposes the metrics.
+      '';
+    };
+
+    telemetryPath = mkOption {
+      type = types.str;
+      default = "/metrics";
+      description = ''
+        Path under which to expose metrics.
+      '';
+    };
+
+    socketPath = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = "/run/unbound/unbound.socket";
+      description = ''
+        Path to the unbound socket
+      '';
+    };
+
+    extraParams = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      example = [ "--log-level" ];
+      description = ''
+        Additional parameters passed to <command>unbound-telemetry</command>.
+      '';
+    };
+  };
+
+  serviceOpts = mkMerge ([{
+    serviceConfig = {
+      # User = "unbound";
+      ExecStart = ''
+        ${pkgs.prometheus-unbound-exporter}/bin/unbound-telemetry \
+          ${cfg.fetchType} \
+          --bind ${cfg.listenAddress}:${toString cfg.port} \
+          --path ${cfg.telemetryPath} \
+          ${optionalString (cfg.socketPath != null) "--control-interface ${cfg.socketPath}"} \
+          ${toString cfg.extraParams}
+      '';
+    };
+  }] ++ [
+    (mkIf config.services.nginx.enable {
+      after = [ "unbound.service" ];
```

Comment:
Why is this only needed if `nginx` is enabled?

Reply:
I copied it from the nginx exporter.